frontend/src/metabase/sharing/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx

```diff
@@ -42,15 +39,18 @@ export const QuestionPublicLinkPopover = ({
       })
     : null;
 
-  const createPublicQuestionLink = async () => {
-    await dispatch(createPublicLink(question.card()));
+  const [createPublicQuestionLink] = useCreateCardPublicLinkMutation();
+  const [deletePublicQuestionLink] = useDeleteCardPublicLinkMutation();
+
+  const handleCreatePublicQuestionLink = async () => {
+    await createPublicQuestionLink({ id: question.id() });
   };
-  const deletePublicQuestionLink = async () => {
+  const handleDeletePublicQuestionLink = async () => {
     trackPublicLinkRemoved({
       artifact: "question",
       source: "public-share",
     });
-    await dispatch(deletePublicLink(question.card()));
+    await deletePublicQuestionLink({ id: question.id() });
```

Comment:
(Just noticed that we bring in the entire card just to delete a public link in the old code, wow.)

Reply:
Yeah it wasn't the best idea 😂